doc/overview.md

```diff
@@ -1,31 +1,69 @@
+The produced Ignition config is very much not intended to be human-friendly. It
+is an artifact produced by ct that users should simply pass along to their
+machines. JSON was chosen over a binary format to make the process more
+transparent and to allow power users to inspect/modify what ct produces, but it
+would have worked fine if the result from ct had not been human readable at all.
 
-### Overview of the config types
+[ignition]: https://github.com/coreos/ignition
 
-#### Container Linux Config
+## Why a two step process?
 
-The Container Linux Config, as the name implies, is specific to Container Linux. It is designed to be easy to read and easy to write, something at which YAML excels. This is the config format that users are intended to directly manipulate. This config format is not explicitly versioned and instead depends on the particular version of ct that is in use. It is assumed that the user will manually keep their Container Linux Configs and ct in sync with one another. Because both the configs and ct are under the users control, this can be an atomic operation and it is unnecessary to explicitly version the Container Linux Config.
+There are a couple factors motivating the decision to not incorporate support
+for Container Linux Configs directly into the boot process of Container Linux
+(as in, the ability to provide a Container Linux Config directly to a booting
+machine, instead of an Ignition config).
 
-Over time, the Container Linux Config will gain more and more support for tasks and configuration specific to instances of Container Linux as well as the environment in which they run.
+First and foremost, by making users run their configs through ct before they
+attempt to boot a machine, issues with their configs can be caught before any
+machine attempts to boot. This will save users time, as they can much more
+quickly find problems with their configs. Were users to provide Container Linux
+Configs directly to machines at first boot, they would need to find a way to
+extract the ignition logs from a machine that may have failed to boot, which can
+be a slow and tedious process.
 
-#### Ignition Config
+Second, YAML parsing is a complex process that in the past has been rather
+error-prone. By only doing JSON parsing in the boot path, we can guarantee that
+the utilities necessary for a machine to boot are simpler and more reliable. We
+want to allow users to use YAML however, as it's much more human-friendly than
+JSON, hence the decision to have a tool separate from the boot path to
+"transpile" YAML configurations to machine-appropriate JSON ones.
 
```

Comment:
It might also be worth mentioning that users don't need to explicitly use ct if they are using matchbox.